core-api/src/main/kotlin/com/jakemoore/datakache/core/connections/mongo/changestream/ChangeStreamEventProcessor.kt

```
@@ -0,0 +1,386 @@
+
+    /**
+     * Closes the event channel and cleans up resources.
+     */
+    @Suppress("RedundantSuspendModifier")
+    suspend fun cleanup() {
+        val channel = eventChannel
+        if (channel != null) {
+            try {
+                channel.close()
+                context.logger.debug("Event channel closed")
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)